agent/agent.go

```diff
@@ -206,6 +206,7 @@ func (a *Agent) run(ctx context.Context) {
 		leaving       = a.leaving
 		subscriptions = map[string]context.CancelFunc{}
 	)
+	defer session.close()
```

Comment:
Ah ok, good point - so wrapping it in a function is still the best idea since that evaluates when the function executes.